src/main/java/org/osc/manager/rest/server/api/PolicyApis.java

```diff
@@ -44,70 +42,71 @@
     private static final Logger logger = Logger.getLogger(PolicyApis.class);
 
     /**
-     * Creates the Policy
+     * Creates the Policy for a given domain
      *
      * @return policy
      */
+    @Path("/{domainId}")
     @POST
-    public PolicyEntity createPolicy(@Context HttpHeaders headers ,PolicyEntity entity) {
+    public PolicyEntity createPolicy(@PathParam("domainId") Long domainId, PolicyEntity entity) {
 
         logger.info("Creating Policy Entity...");
         //TODO Sudhir: Add db calls here
-        return entity;
+        return null;
     }
 
     /**
-     * Updates the Policy
+     * Updates the Policy for a given domain
      *
      * @return - updated policy
      */
-    @Path("/{policyId}")
+    @Path("/{domainId}")
```

Comment:
actually the policy id should also be in the url, typically for update, delete and get the targeted object id should be in the url as well. So the resource path should be: /domains/{domainid}/policies/{policyid} . See this example https://github.com/opensecuritycontroller/osc-core/blob/master/osc-rest-server/src/main/java/org/osc/core/broker/rest/server/api/VirtualizationConnectorApis.java#L343 where the security groups resource is under the virtualization connectors resource and follow the same pattern for this.